### PR TITLE
Add security role data source

### PR DESCRIPTION
**Pull request by k-yomo**
Resolves https://github.com/elastic/terraform-provider-elasticstack/issues/82

This PR adds security role data-source in reference to existing security role resource.
https://github.com/elastic/terraform-provider-elasticstack/blob/main/internal/elasticsearch/security/role.go

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 ### Added
 - New resource `elasticstack_elasticsearch_logstash_pipeline` to manage Logstash pipelines ([Centralized Pipeline Management](https://www.elastic.co/guide/en/logstash/current/logstash-centralized-pipeline-management.html)) ([#151](https://github.com/elastic/terraform-provider-elasticstack/pull/151))
+- Add`elasticstack_elasticsearch_security_role` data source ([#177](https://github.com/elastic/terraform-provider-elasticstack/pull/177))
```
Comment (tobio) [inline CHANGELOG.md]:
```suggestion
- Add `elasticstack_elasticsearch_security_role` data source ([#177](https://github.com/elastic/terraform-provider-elasticstack/pull/177))
```

File: docs/data-sources/elasticsearch_security_role.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
+---
+# generated by https://github.com/hashicorp/terraform-plugin-docs
+page_title: "elasticstack_elasticsearch_security_role Data Source - terraform-provider-elasticstack"
+subcategory: ""
```
Comment (tobio) [inline docs/data-sources/elasticsearch_security_role.md]:
```suggestion
subcategory: "Security"
```

**Review by tobio — APPROVED** [review]
Couple of small nits. LGTM

File: templates/data-sources/elasticsearch_security_role.md.tmpl [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+subcategory: "Security"
+layout: ""
+page_title: "Elasticstack: elasticstack_elasticsearch_security_role Data Source"
+description: |-
+  Retrieves roles in the native realm.
+---
+
+# Data Source: elasticstack_elasticsearch_security_role
+
+Use this data source to  information about existing Elasticsearch role. See, https://www.elastic.co/guide/en/elasticsearch/reference/current/security-api-get-role.html
```
Comment (tobio) [inline templates/data-sources/elasticsearch_security_role.md.tmpl]:
```suggestion
Use this data source to get information about an existing Elasticsearch role. See, https://www.elastic.co/guide/en/elasticsearch/reference/current/security-api-get-role.html
```

Maybe there's other words you intended to have here, happy to use those instead too :)

Comment (k-yomo) [reply templates/data-sources/elasticsearch_security_role.md.tmpl]: Whoops, I guess I just deleted it by mistake, fixed🙏
https://github.com/elastic/terraform-provider-elasticstack/pull/177/commits/6ec61a462cd1f681df1ff47ae1208775ceadbc5d

**Comment by k-yomo** [comment]
Ah, I'll fix merge conflict after another PR gets merged

**Review by tobio — APPROVED** [review]

**Pull request merged**